### ISSUE TITLE
Misrepresented WRNs

### ISSUE THREAD

**Issue by DewaldHoman** [google-research/fixmatch#36]
Hi there, I really enjoy your work. However, I feel a bit like standard WRNs have been misrepresented in a way throughout the _ixMatch lifespan. In MixMatch Section 4.2.2: '135 filters per layer' does not make sense to me. Firstly, something like '135 filters for the first residual block' would be more justifiable. Secondly, for a standard WRN starting with 16 filters, 135 filters would mean a k-value of 8.4375, as in WRN28-8.4375.

Furthermore, ReMixMatch refers to 'the same WRN-37-2-network' in Section 4.2, but in reality this is a different model that came about inbetween the papers. My problem lies not primarily in the fact that 135 was changes to 128 filters (the k-value of 8 seems more reasonable: e.g. WRN-28-8). I do however feel a bit uneasy that it was presented as a WRN-37-2 model. Moreover, FixMatch also refers to the same WRN-37-2 model. Indeed, it could have 37 conv layers and a similar structure to a WRN. But to my knowledge, such a model does not exist and the closest standard WRN models have either 34 or 40 conv layers (why not use these instead?). Using anything other than `--scales=3` would defiate from the standard WRN model structure and the resulting model would be another creature.

I understand that the focus of this research is on the SSL-methods and the comparison thereof using the same models. It might also be that my comprehension in the above is lacking. I would however appreciate clarity surrounding issues like these, rather than avoiding or dragging along underlying deficiencies if any are known. Thanks.

**Comment by carlini** [comment]
Thanks for the comments. Yeah, we were definitely less precise here than we should have been. MixMatch and ReMixMatch are published papers so they'll have to stay as they are, but for FixMatch we'll try and clarify the model description.

As you say, there is no standard for exactly what "WRN-37-2" means---this is by design. We didn't want to mislead people into thinking that we were using one model when we were using another, so we picked the name that was the closest approximation to describe what we were doing while not conflicting with any other names. The reason we switched from 135 to 128 filters was to try and make it more WRN compatible. Initially we picked 135 filters because that's how many got us closest in parameter count to prior work that did CIFAR-100.

We'll try to describe this in more detail somewhere in the paper, but because we're trying to focus on what's new and interesting in eight pages (semi-supervised learning, in this case) we leave out details of architecture that are common between all methods. Importantly, this is why we've always released code simultaneous with the paper: it will never be possible to completely describe experimental setup, and having the actual code allows the interested reader to read these details.

Note that because we implement all methods with the same framework and the same models, differences in architecture won't impact the relative results. So while switching from our resnet to something more standard might slightly change accuracy numbers, we shouldn't expect one method to suddenly outperform another.
